rockhound/mcmurray_facies.py

```diff
@@ -0,0 +1,100 @@
+     - electronic data; Alberta Research Council, ARC/AGS Special Report 6._
+
+    Please go to the links below for more information and the dataset:
+
+    Report for Athabasca Oil Sands Data McMurray/Wabiskaw Oil Sands Deposit
+    http://ags.aer.ca/document/OFR/OFR_1994_14.PDF
+
+    Electronic data for Athabasca Oil Sands Data McMurray/Wabiskaw Oil Sands Deposit
+    http://ags.aer.ca/publications/SPE_006.html
+
+    In the metadata file
+    https://github.com/JustinGOSSES/MannvilleGroup_Strat_Hackathon/blob/master/
+    SPE_006_originalData/Metadata/SPE_006.txt -> SPE_006.txt.
+    the dataset is described as Access Constraints: Public and Use Constraints:
+    Credit to originator/source required. Commercial reproduction not allowed.
+
+    _The Latitude and longitude of the wells is not in the original dataset.
+    https://github.com/dalide -> @dalide used the Alberta Geological Society’s UWI
+    conversion tool to find lat/longs for each of the well UWIs. A CSV with the
+    coordinates of each well’s location can be found
+    https://github.com/JustinGOSSES/MannvilleGroup_Strat_Hackathon/blob/master/well_lat_lng.csv.
+    These were then used to find each well’s nearest neighbors.
+
+    Please note that there are a few misformed .LAS files in the full dataset, so the code
+    in this repository skips those.
+
+    If for some reason the well data is not found at the links above, you should
+    be able to find it
+    https://github.com/JustinGOSSES/MannvilleGroup_Strat_Hackathon/tree/master/SPE_006_originalData
+
+    If the file isn't already in your data directory, it will be downloaded
+    automatically.
+
+    Parameters
+    ----------
+    load : bool
+        Wether to load the data into a :class:`pandas.DataFrame` or just return the
+        path to the downloaded data.
+
+    Returns
+    -------
+    mcmurray_facies : :class:`pandas.DataFrame` or str
+        The loaded data or the file path to the downloaded data.
+        The :class:`pandas.DataFrame` contains the following data:
+
+        columns : ['CALI', 'COND', 'DELT', 'DEPT', 'DPHI', 'DT', 'GR', 'ILD', 'ILM',
+        'NPHI', 'PHID', 'RHOB', 'SFL', 'SFLU', 'SN', 'SP', 'UWI', 'SitID','lat', 'lng',
+         'Depth', 'LithID', 'W_Tar', 'SW', 'VSH', 'PHI', 'RW','lithName']
+    """
+
+    fname = REGISTRY.fetch("mcmurray_facies_v1.tar.gz")
+    if not load:
+        return fname
+
+    ##### Insert here all the code to load to take the unzipped files and process into
+    ##### single dataframe for the called dataset
+    ##### These try and excepts are here as some versions of pandas don't work with
+    ##### generic buffers, which fname may be depending on your OS.
+    ##### We get around this by saving as CSV zipped via tar with xz compression.
+    try:
+        test_test = tarfile.open(fname, "r:xz")
+        test_test.extractall("mcmurray_facies_v1")
+        data = pd.read_csv("mcmurray_facies_v1/mcmurray_facies_v1.csv", engine="python")
```

Comment:
Use the `Untar` processor instead: https://www.fatiando.org/pooch/latest/api/generated/pooch.Untar.html#pooch.Untar

Reply:
Changed